### PR TITLE
chore: webpack examples

### PR DESCRIPTION
**Pull request by edwardnyc**
cherry-pick from manualChunk branch

### CODE REVIEW

File: examples/webpack/env.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/// <reference types="vite/client" />
```
Comment (posva) [inline examples/webpack/env.d.ts:1]: this looks like a leftover

File: examples/webpack/env.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+declare module '@vue-router/routes' {
```
Comment (posva) [inline examples/webpack/env.d.ts:3]: I think we can remove this part as it's already tested in the vite playground. Let's keep this example to the minimal to test out locally 💯

File: examples/webpack/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "webpack",
+  "version": "0.1.0",
```
Comment (posva) [inline examples/webpack/package.json:3]: no need

```suggestion
```

File: examples/webpack/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  "private": true,
+  "scripts": {
+    "dev": "vue-cli-service serve",
+    "build": "vue-cli-service build",
+    "lint": "vue-cli-service lint"
+  },
+  "dependencies": {
+    "unplugin-vue-router": "workspace:*",
```
Comment (posva) [inline examples/webpack/package.json:11]: move to devDep

File: examples/webpack/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    "vue": "^3.2.13",
+    "vue-router": "^4.1.0"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.0.0",
```
Comment (posva) [inline examples/webpack/package.json:16]: let's remove this dependency unless completely necessary. it's quite heavy

File: examples/webpack/public/index.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html lang="">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <link rel="icon" href="<%= BASE_URL %>favicon.ico">
```
Comment (posva) [inline examples/webpack/public/index.html:7]: let's remove the favicon and this line

**Review by posva — CHANGES_REQUESTED** [review]
Can you rebase against `main` and run `pnpm run lint --write` to run the formatting?

File: examples/webpack/package.json (hunk 5 above)
Comment (edwardnyc) [reply examples/webpack/package.json:16]: As you suggested I have removed,  but removing this dependency will pop up a warning during installation.

image

Comment (posva) [reply examples/webpack/package.json]: Maybe it's worth basing it in a new project:

Screenshot 2022-07-14 at 10 13 28

File: examples/webpack/src/App.vue [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        <RouterLink to="/HelloWorld">HelloWorld</RouterLink>
+      </nav>
+    </div>
+  </header>
+  <router-view></router-view>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import HelloWorld from './pages/HelloWorld.vue'
```
Comment (posva) [inline examples/webpack/src/App.vue:19]: let's remove this component completely

File: examples/webpack/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
+{
+  "extends": "@vue/tsconfig/tsconfig.web.json",
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@/*": ["./src/*"]
+    }
+  },
+  "include": [
```
Comment (posva) [inline examples/webpack/tsconfig.json:9]: this can be simplified

**Review by posva — APPROVED** [review]
Thanks for this! I added a couple of things and pushed to you PR afte rebasing. Nor sure why the author got overriden 🤔

**Pull request merged**